### PR TITLE
Add clean cmake build and some improvements

### PR DESCRIPTION
**Pull request by adydychk**
Implemented:

- [x] Stable clean cmake build
- [x] Clean cmake build in 2 configs
- [ ] Product_configs unified
- [x] Verbose: strings
- [x] Verbose: Artifacts listing
- [x] Verbose: gcc --version
- [ ] (?) Update developer\`s documentation
- [ ] Release/ release / debug options?

NB: Need to merge with this Pull request: ...

### CODE REVIEW

File: conf_linux_public.py [hunk 1]
```diff
@@ -115,7 +126,7 @@ def _get_api_version(repo_path):
             },
             {
                 'path': 'plugins.cfg',
-                'pack_as': 'bin/release/plugins.cfg'
+                'pack_as': 'bin/Release/plugins.cfg'
```
Comment (azhogov) [inline conf_linux_public.py]: incorrect, no changes should be here

File: conf_linux_public_api_latest.py [hunk 2]
```diff
@@ -87,16 +87,23 @@ def _get_api_version(repo_path):
 action('count api version and build number',
        callfunc=(set_env, [MEDIA_SDK_REPO_DIR], {}))
 
-CMAKE_CFG = 'intel64.make.' + options.get('BUILD_TYPE')
-options['BUILD_DIR'] = MEDIA_SDK_REPO_DIR / '__cmake' / CMAKE_CFG
-
-
 action('compiler version',
-       cmd=f'{ENABLE_DEVTOOLSET} && gcc --version')
+       cmd=f'{ENABLE_DEVTOOLSET} && gcc --version',
```
Comment (azhogov) [inline conf_linux_public_api_latest.py]: please delete this config and use args

File: conf_linux_public.py [hunk 3]
```diff
@@ -87,16 +87,22 @@ def _get_api_version(repo_path):
...
+
+cmake_command = ['cmake',
+                 '--no-warn-unused-cli"',
+                 '"-Wno-dev -G "Unix Makefiles"',
+                 '-DWARNING_FLAGS="-Wall -Werror"',
+                 '-DCMAKE_C_FLAGS_RELEASE="-O2 -D_FORTIFY_SOURCE=2 -fstack-protector"',
+                 '-DCMAKE_CXX_FLAGS_RELEASE="-O2 -D_FORTIFY_SOURCE=2 -fstack-protector"',
+                 str(MEDIA_SDK_REPO_DIR),
+]
+cmake_string = ' '.join(cmake_command)
```
Comment (azhogov) [inline conf_linux_public.py]: strange name and transformation

File: conf_linux_public_api_latest.py (hunk 2 above)
Comment (adydychk) [reply conf_linux_public_api_latest.py]: What is the reason?

**Pull request merged**